modules/custom/activity_creator/js/notifications.js

```diff
@@ -4,30 +4,36 @@
  */
 
 (function ($) {
-    /**
-     * Behaviors.
-     */
-    Drupal.behaviors.notificationUpdate = {
-        attach: function (context, settings) {
-            // TODO Implement this with a no-js fallback.
-            var notification_count = $('.notification-bell .badge');
+  /**
+   * Notification centre bell update behavior.
+   */
+  Drupal.behaviors.notificationUpdate = {
+    attach: function (context, settings) {
 
-          if (notification_count.val() != "0") {
-              $('.notification-bell').click(function(e) {
-                  $.ajax({
-                      method: 'POST',
-                      url: '/ajax/notifications-mark-as-read',
-                      data: { },
-                      success: function(result) {
-                          // Update the notification bell.
-                          var remaining_notifications = result['remaining_notifications'];
+      $('.notification-bell', context).once('notificationUpdate').click(function(e) {
```

Comment:
We should move this callback to its own property. The callback always updates all the notification counts on the page (both desktop and mobile) and the same callback can be used for all notification bells that may trigger this (no need to create new functions).